### PR TITLE
Expression

### PR DESCRIPTION
**Pull request by AndHager**
Implementation of: arguments, boolean expressions, boolean operations, comparisons, comprehensions, lambda (no condition) and operators.

### CODE REVIEW

File: src/main/kotlin/io/github/oxisto/reticulated/Pair.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019, Fraunhofer AISEC. All rights reserved.
```
Comment (oxisto) [inline src/main/kotlin/io/github/oxisto/reticulated/Pair.kt]: Can you update the copyright to 2020?

Comment (AndHager) [reply src/main/kotlin/io/github/oxisto/reticulated/Pair.kt]: Yes of course.

File: src/main/kotlin/io/github/oxisto/reticulated/Pair.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2020, Fraunhofer AISEC. All rights reserved.
```
Comment (oxisto) [inline src/main/kotlin/io/github/oxisto/reticulated/Pair.kt]: I suggest the following copyright header:
```
* Copyright (c) 2020, Christian Banse and Andreas Hager. All rights reserved.
```

File: src/main/kotlin/io/github/oxisto/reticulated/ast/expression/boolean_ops/BaseBooleanOp.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package io.github.oxisto.reticulated.ast.expression.boolean_ops
```
Comment (oxisto) [inline src/main/kotlin/io/github/oxisto/reticulated/ast/expression/boolean_ops/BaseBooleanOp.kt]: I would avoid `_` in the package name

**Comment by oxisto** [comment]
Hmm I am not sure that I understand the current state of the boolean expressions but it seems to handle the way it is also documented in the language reference.

File: src/main/kotlin/io/github/oxisto/reticulated/Pair.kt (hunk 2 above)
Comment (AndHager) [reply src/main/kotlin/io/github/oxisto/reticulated/Pair.kt]: Done commit e01b9e2189b76694cb29ff42737780b701664745

File: src/main/kotlin/io/github/oxisto/reticulated/ast/expression/boolean_ops/BaseBooleanOp.kt (hunk 3 above)
Comment (AndHager) [reply src/main/kotlin/io/github/oxisto/reticulated/ast/expression/boolean_ops/BaseBooleanOp.kt]: Done with commit dade646

**Pull request merged**